docs: fix description of treemanager api (#1135)
Refinitiv/refinitiv-ui: documents/src/pages/elements/tree.md

});
```

## Manipulating item properties

> Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`.

Reviewer: ```suggestion
Item properties of Tree could be read and updated programmatically through its [Tree Manager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`.
```

```javascript
// Select the item which value is '1.1'
const tree = document.querySelector('ef-tree');
tree.data = [
  {
    label: 'Group',
    value: '1.0',
    items: [{
      label: 'Item 1.1',
      value: '1.1'
    },
    {
      label: 'Item 1.2',
      value: '1.2'
    }]
  }
];
console.log(tree.values); // Expected output: []

const treeNodes = tree.manager.getTreeNodes();

> const selectedNode = treeNodes.find(treeNode => treeNode.value === '1.1');

Reviewer: `selectedNode` is rather misleading: the node is not selected yet.
```suggestion
const node = treeNodes.find(treeNode => treeNode.value === '1.1');
```

selectedNode.selected = true;
console.log(tree.values); // Expected output: ['1.1']
```

```typescript
// Select the item which value is '1.1'
const tree = document.querySelector('ef-tree')!;

> tree.data = [

Reviewer: `data` would be an unknown property without proper type def. Missing `import type { Tree } from '@refinitiv-ui/elements/tree'`?

Author: Fixed.

  {
    label: 'Group',
    value: '1.0',
    items: [{
      label: 'Item 1.1',
      value: '1.1'
    },
    {
      label: 'Item 1.2',
      value: '1.2'
    }]
  }
];
console.log(tree.values); // Expected output: []

const treeNodes = tree.manager.getTreeNodes();

> const selectedNode = treeNodes.find(treeNode => treeNode.value === '1.1');

Reviewer: `selectedNode` is rather misleading: the node is not selected yet.
```suggestion
const node = treeNodes.find(treeNode => treeNode.value === '1.1');
```

selectedNode.selected = true;
console.log(tree.values); // Expected output: ['1.1']
```

## Custom renderer

> Tree supports custom rendering by providing a renderer property that can define how each item is displayed. The renderer is a callback function that takes a data item, a Collection Composer, and previously mapped item elements (if any), and must return an HTMLElement. You can use TreeNode to provides access to the current item's data property. Note that for performance sensitive use cases such as a large number of items, consider using [Collection Composer](/custom-components/utils/data-management#collection-composer) instead.

Reviewer: ```suggestion
Tree defines how each of its item is displayed with `renderer` property. You can customise this renderer by setting a callback function to the property. [Tree Node](/custom-components/utils/tree-node) is the easiest way to implement the function. Note that for performance sensitive use cases such as a large number of items, consider using [Collection Composer](/custom-components/utils/data-management#collection-composer) instead.
```

```javascript
import { uuid } from '@refinitiv-ui/utils/uuid.js';
import { CheckedState, TreeManager, TreeManagerMode } from '@refinitiv-ui/elements/tree';

> // Implement Tree's default render with Tree Node instead of Collection Composer

Reviewer: ```suggestion
// Implement Tree's default renderer with Tree Node instead of Collection Composer
```

// for comparison, check https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/tree/helpers/renderer.ts
const createTreeRenderer = (context) => {
  const key = uuid();

  return (item, composer, element = document.createElement('ef-tree-item')) => {
    const multiple = context?.multiple === true;
    const noRelation = context?.noRelation === true;
    const mode = !multiple || !noRelation ? TreeManagerMode.RELATIONAL : TreeManagerMode.INDEPENDENT;
    const manager = context?.manager || context?.treeManager || new TreeManager(composer, mode);

    const treeNode = manager.getTreeNode(item);
    element.multiple = multiple;
    element.item = item;
    element.id = `${key}-${item.value || ''}`;
    element.depth = treeNode.getDepth();
    element.parent = treeNode.isParent();
    element.expanded = treeNode.expanded;
    element.checkedState =
      !multiple && element.parent ? CheckedState.UNCHECKED : treeNode.getCheckedState();
    element.icon = treeNode.icon;
    element.label = treeNode.label;
    element.disabled = treeNode.disabled;
    element.readonly = treeNode.readonly;
    element.highlighted = treeNode.highlighted;

    return element;
  };
};

> tree.renderer = createTreeRenderer(this)

Reviewer: ```suggestion
tree.renderer = createTreeRenderer(tree)
```

```

```typescript
import { CheckedState, TreeManager, TreeManagerMode } from '@refinitiv-ui/elements/tree';
import { uuid } from '@refinitiv-ui/utils/uuid.js';
import type { CollectionComposer } from '@refinitiv-ui/utils/collection.js';
import type { TreeDataItem, TreeItem } from '@refinitiv-ui/elements/tree';

type RendererScope<T extends TreeDataItem> = {
  multiple?: boolean;
  noRelation?: boolean;
  manager?: TreeManager<T>;
  treeManager?: TreeManager<T>;
};

> // Implement Tree's default render with Tree Node instead of Collection Composer

Reviewer: ```suggestion
// Implement Tree's default renderer with Tree Node instead of Collection Composer
```

// for comparison, check https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/tree/helpers/renderer.ts
export const createTreeRenderer = <T extends TreeDataItem = TreeDataItem>(
  context?: unknown
): ((item: T, composer: CollectionComposer<T>, element?: HTMLElement) => HTMLElement) => {
  const key: string = uuid();

  return (
    item: T,
    composer: CollectionComposer<T>,
    element: HTMLElement = document.createElement('ef-tree-item')
  ): HTMLElement => {
    const _context = context as RendererScope<T> | undefined;
    const _element = element as TreeItem;
    const multiple = _context?.multiple === true;
    const noRelation = _context?.noRelation === true;
    const mode = !multiple || !noRelation ? TreeManagerMode.RELATIONAL : TreeManagerMode.INDEPENDENT;

>     const manager = _context?.manager || new TreeManager(composer, mode);

Reviewer: ```suggestion
    const manager = _context?.manager || _context?.treeManager || new TreeManager(composer, mode);
```

    const treeNode = manager.getTreeNode(item);
    _element.multiple = multiple;
    _element.item = item;
    _element.id = `${key}-${item.value || ''}`;
    _element.depth = treeNode.getDepth();
    _element.parent = treeNode.isParent();
    _element.expanded = treeNode.expanded;
    _element.checkedState =
      !multiple && _element.parent ? CheckedState.UNCHECKED : treeNode.getCheckedState();
    _element.icon = treeNode.icon;
    _element.label = treeNode.label;
    _element.disabled = treeNode.disabled;
    _element.readonly = treeNode.readonly;
    _element.highlighted = treeNode.highlighted;

    return _element;
  };
};

> tree.renderer = createTreeRenderer(this)

Reviewer: ```suggestion
tree.renderer = createTreeRenderer(tree)
```

});
```

> ## Manipulate item property

Reviewer: Let's refer to Tree Select's feedback.

Author: Fixed.

});
```

## Manipulating item properties

> Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. 

Reviewer: Let's cleanup trailing whitespaces.

> Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. 

Reviewer: ```suggestion
Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. In adiition, they could be used for [custom renderer](./custom-components/utils/tree-node#custom-renderer) too. 
```
```

> Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. In adiition, they could be used for [custom renderer](./custom-components/utils/tree-node#custom-renderer) too.

Reviewer: ```suggestion
Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. In addition, they could be used for [custom renderer](./custom-components/utils/tree-node#custom-renderer) too.
```

> Item properties of Tree could be read and updated programmatically through its [TreeManager](./custom-components/utils/tree-manager) which is available as `manager` property. Retrieve [TreeNode(s)](./custom-components/utils/tree-node) representing each item by calling `getTreeNode()` or `getTreeNodes()` of `manager`. In addition, they could be used for [custom renderer](./custom-components/utils/tree-node#custom-renderer) too.

Reviewer 2: `In addition, they could be used for [custom renderer](./custom-components/utils/tree-node#custom-renderer) too.`

I feel that this content should be removed from this section. What we could do is to introduce another section, 'Custom Renderer' so we can provide completed guideline.

Reviewer: Let's move "Custom Renderer" from "Tree Node" to "Tree" & "Tree Select" + revise it content a little as readers focus on Component first rather than "Tree Node" first.

Author: Moved and wrote new Custom renderer section on both components.

```javascript
const tree = document.querySelector('ef-tree');
tree.data = [
  {
    label: 'Group',
    value: '1.0',
    items: [{
      label: 'Item 1.1',
      value: '1.1'
    },
    {
      label: 'Item 1.2',
      value: '1.2'
    }]
  }
];

> const node0 = tree.manager.getTreeNodes()[0]; // get the item at index number 0.

Reviewer: naming variable with numbers looks sloppy.
```suggestion
const firstNode = tree.manager.getTreeNodes()[0]; // get the item at index number 0.
```

node0.label = 'Group 1'; // edit label property which this will reflect to the component.
```

```typescript

> const tree = document.querySelector('ef-tree')!;    

Reviewer: let's cleanup trailing whitespaces.

Author: Updated.

tree.data = [
  {
    label: 'Group',
    value: '1.0',
    items: [{
      label: 'Item 1.1',
      value: '1.1'
    },
    {
      label: 'Item 1.2',
      value: '1.2'
    }]
  }
];

> const node0 = tree.manager.getTreeNodes()[0]; // get the item at index number 0.

Reviewer: naming variable with numbers looks sloppy.
```suggestion
const firstNode = tree.manager.getTreeNodes()[0]; // get the item at index number 0.
```
